pkg/controller/certificate-shim/sync.go

```diff
@@ -508,6 +508,106 @@ func certNeedsUpdate(a, b *cmapi.Certificate) bool {
 		return true
 	}
 
+	if a.Spec.RevisionHistoryLimit != b.Spec.RevisionHistoryLimit {
+		return true
+	}
+
+	aAlgorithm, bAlgorithm := cmapi.RSAKeyAlgorithm, cmapi.RSAKeyAlgorithm
+	if a.Spec.PrivateKey != nil && a.Spec.PrivateKey.Algorithm != "" {
+		aAlgorithm = a.Spec.PrivateKey.Algorithm
+	}
+
+	if b.Spec.PrivateKey != nil && b.Spec.PrivateKey.Algorithm != "" {
+		bAlgorithm = b.Spec.PrivateKey.Algorithm
+	}
+
+	if aAlgorithm != bAlgorithm {
+		return true
+	}
+
+	aEncoding, bEncoding := cmapi.PKCS1, cmapi.PKCS1
```

Comment:
Suggestion: (same as above with the same reasoning) `var aEncoding, bEncoding cmapi.PrivateKeyEncoding`